pkg/asset/machines/aws/machinesets.go

```diff
@@ -32,28 +34,62 @@ func MachineSets(clusterID string, region string, subnets map[string]string, poo
 		if int64(idx) < total%numOfAZs {
 			replicas++
 		}
-
-		subnet, ok := subnets[az]
-		if len(subnets) > 0 && !ok {
+		subnet, ok := (*subnets)[az]
+		if len(*subnets) > 0 && !ok {
 			return nil, errors.Errorf("no subnet for zone %s", az)
 		}
-		provider, err := provider(
-			clusterID,
-			region,
-			subnet,
-			mpool.InstanceType,
-			&mpool.EC2RootVolume,
-			mpool.EC2Metadata,
-			mpool.AMIID,
-			az,
-			role,
-			userDataSecret,
-			userTags,
-		)
+
+		privateSubnet := !subnet.Public
+		instanceType := mpool.InstanceType
+		nodeLabels := make(map[string]string, 3)
+		nodeTaints := []corev1.Taint{}
+
+		if pool.Name == types.MachinePoolEdgeRoleName {
+			// edge pools typically do not receive the same workloads between
+			// different zoneGroups, thus the installer will discover preferred
+			// instance based on the installer's preferred instance lookup.
+			if subnet.PreferredEdgeInstanceType != "" {
+				instanceType = subnet.PreferredEdgeInstanceType
+			}
+			nodeLabels = map[string]string{
+				"node-role.kubernetes.io/edge": "",
+				"zone_type":                    subnet.ZoneType,
+				"zone_group":                   subnet.ZoneGroup,
```

Comment:
That's a good question. I didn't hear any suggestions like that on the EP. It could be changed, but I wonder if we could have some restrictions or conventions for using it. Any suggestion?

Reply:
I think namespacing under machine.openshift.io would probably make sense, but we may already have labels that are similar

One convention is not to use underscores, it would be more idiomatic to use a hyphen.

Remind me what the zone type and groups are again? Are they anything consumed by Machine API or is this just grouping of MachineSets?

Reply:
> Remind me what the zone type and groups are again? Are they anything consumed by Machine API or is this just grouping of MachineSets?

They are not consuming internally (for any OCP component). It was created to group the nodes (created by MachineSets) helping customers to create their applications and selecting workloads for each location in LZ.

Both are attributes from the [`AvailabilityZone` structure](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AvailabilityZone.html):

- `zone_type`: Availability Zone type, in this case, will always be `local-zone`
- `zone_group`: Each zone has its own group. The border group of Local Zone is the metropolitan region to which the zone belongs. For example in Los Angeles, there are two local zones, `us-west-2-lax-1a` and `us-west-2-lax-1b`, both are part of the border group `us-west-2-lax-1`

```
$ aws --region us-west-2 ec2 describe-availability-zones | jq -r '.AvailabilityZones[] | select(.GroupName == "us-west-2-lax-1")'
{
  "State": "available",
  "OptInStatus": "opted-in",
  "Messages": [],
  "RegionName": "us-west-2",
  "ZoneName": "us-west-2-lax-1a",
  "ZoneId": "usw2-lax1-az1",
  "GroupName": "us-west-2-lax-1",
  "NetworkBorderGroup": "us-west-2-lax-1",
  "ZoneType": "local-zone",
  "ParentZoneName": "us-west-2b",
  "ParentZoneId": "usw2-az2"
}
{
  "State": "available",
  "OptInStatus": "opted-in",
  "Messages": [],
  "RegionName": "us-west-2",
  "ZoneName": "us-west-2-lax-1b",
  "ZoneId": "usw2-lax1-az2",
  "GroupName": "us-west-2-lax-1",
  "NetworkBorderGroup": "us-west-2-lax-1",
  "ZoneType": "local-zone",
  "ParentZoneName": "us-west-2d",
  "ParentZoneId": "usw2-az4"
}
```

Reply:
@JoelSpeed [I fixed](patch) the labels added to the machine.openshift.io namespace. IMHO it's a great idea, thanks for pointing it out. Let me know if it fits what you were thinking.

[patch]:https://github.com/openshift/installer/compare/72f58ca741fe1262dbf79c4acef6b138a67af89a..ee094569c1e9b59a55d694f6d83811e57c1c8b87